### DISCUSSION TITLE
Example using React and Go?

### DISCUSSION THREAD

**Discussion by Anan7Codes** [authgear/authgear-server#4276] [Get Help] [answered]
A bit new to using an IAM tool and hence was wondering the right approach to go forward with if I want to integrated Authgear in my apps using React(Custom UI maybe?) and Golang(Fiber).

**Comment by fungc-io** [comment, marked as answer]
Hi @Anan7Codes , Thanks for your message.

You can integrate Authgear into your React app using our SDK. In the frontend, trigger the user authentication and exchange for tokens with the functions provided by the SDK. Then your app can access restricted resources in your Golang backend server by including the access token. The access token proves the requests are sent from an authenticated user.

- Detailed guide can be found here: https://docs.authgear.com/get-started/single-page-app/react
- Here is a sample React project: https://github.com/authgear/authgear-example-react

In the Golang server, you can verify if the user's requests by validating the access token, which is a JWT. 
You can see the Go example code in this page to understand how to decode and validate the token in your service: https://docs.authgear.com/get-started/backend-api/jwt

After validating the JWT and decoding the user, your server can respond to the request based on the authentication status of the user.

It's recommended for developers to start with the default AuthUI for the signup/login screens, as it's easy to integrate and all authentication methods are supported built-in. Once you integrated Authgear on the frontend and backend, and can authenticate the users, you can look into implementing your own custom UI.

Hope it helps, looking forward to your reply.

**Reply by Anan7Codes** [reply]
Great thanks. Will give it a shot.
